rpc/http.go

```diff
@@ -237,6 +238,13 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// until EOF, writes the response to w, and orders the server to process a
 	// single request.
 	ctx := r.Context()
+
+	// The context might be cancelled if the client's connection was closed while waiting for ServeHTTP.
+	if ctx.Err() != nil {
```

Comment:
1. next pattern has better performance
```
select {
case <-ctx.Done(): 
     // handle ctx.Err()
default:
}
```
2. `return` here means 200 HTTP code. Are you sure we don't need write another code? Maybe let's check `geth's` rpc client.go - what http codes are supported there and make something compatible. (j
3. I'm not sure that  you sure `ctx.Err() != nil` means "client gone" - maybe: 
3.1 our local timeout reached and client connection is still alive.
3.2. Ctrl+C pressed on server and all contexts get canceled but all connections are alive yet. Example response code from another server: https://github.com/caddyserver/caddy/blob/d129ae6aec6af2182217ee8a235f4df8cd2bbfde/modules/caddyhttp/server.go#L590 or   https://github.com/golang/go/blob/c5698e31552953a0487ec23e0a015c02067065d3/src/net/http/server.go#L3618 or https://github.com/sethvargo/go-limiter/blob/1a742f09dc99a5f75cccd2bf64e56e6604d780cf/httplimit/middleware.go#L105

Reply:
let's move this piece to separated PR - because seems it's more controversial then everything else.

Reply:
1. Added `libcommon.FastContextErr` for the pattern.
2. Added HTTP 503 return.
3. When I tested this, I saw that `ctx == context.Background()` here and it is coming from the HTTP framework. I might have missed smth, but I haven't found any timeout on this. Likewise I'm not sure that it inherits our rootCtx (which might be another issue for a proper shutdown). I'm not 100% sure about this, so I've removed the log message and added a TODO for improving it later in a separate PR.